src/binary.ts

```diff
@@ -1,36 +1,110 @@
 /* eslint-disable func-style */
 
 import { coreTypes } from "./types";
-const { HashPrefix } = require("./hash-prefixes");
-const { BinaryParser } = require("./serdes/binary-parser");
-const { BinarySerializer, BytesList } = require("./serdes/binary-serializer");
-const { bytesToHex, slice, parseBytes } = require("./utils/bytes-utils");
+import { BinaryParser } from "./serdes/binary-parser";
+import { AccountID } from "./types/account-id";
+import { HashPrefix } from "./hash-prefixes";
+import { BinarySerializer, BytesList } from "./serdes/binary-serializer";
+import { sha512Half, transactionID } from "./hashes";
+import { FieldInstance } from "./enums";
+import { STObject } from "./types/st-object";
 
-const { sha512Half, transactionID } = require("./hashes");
+/**
+ * Construct a BinaryParser
+ *
+ * @param bytes hex-string to construct BinaryParser from
+ * @returns A BinaryParser
+ */
+const makeParser = (bytes: string): BinaryParser => new BinaryParser(bytes);
```

Comment:
why not just use a constructor?

Reply:
Honestly have no clue. If it was me writing it from scratch I wouldn't bother. But this was from the previous iteration of the design, and I didn't want to change too much of what was being exported from files.

I could take it or leave it, I don't think it's a huge deal either way.